Shoko.Server/API/v3/Controllers/RenamerController.cs

```diff
@@ -13,39 +14,204 @@ namespace Shoko.Server.API.v3.Controllers
+        /// </summary>
+        /// <param name="renamerID">Renamer ID</param>
+        /// <returns></returns>
+        [HttpGet("{renamerID}")]
+        public ActionResult<Models.Shoko.Renamer> GetRenamer([FromRoute] int renamerID)
+        {
+            var renamer = RenameFileHelper.Renamers.Values.FirstOrDefault(r => r.ID == renamerID);
+            if (renamer == null)
+                return NotFound("Renamer not found.");
+
+            return new Models.Shoko.Renamer(renamer);
+        }
+
+        /// <summary>
+        /// Disable the <see cref="Models.Shoko.Renamer"/> by the given <paramref name="renamerID"/>.
+        /// </summary>
+        /// <param name="renamerID">Renamer ID</param>
+        /// <returns></returns>
+        [Authorize("admin")]
+        [HttpPost("{renamerID}/Disable")]
+        public ActionResult DisableRenamerByRenamerID(int renamerID)
+        {
+            var renamer = RenameFileHelper.Renamers.Values.FirstOrDefault(r => r.ID == renamerID);
+            if (renamer == null)
+                return NotFound("Renamer not found.");
+
+            ServerSettings.Instance.Plugins.EnabledRenamers[renamer.Name] = false;
+            ServerSettings.Instance.SaveSettings();
+
+            return Ok();
         }
 
-        [HttpDelete("{renamerID}")]
-        public ActionResult Disable(string renamerID)
+        /// <summary>
+        /// Enable the <see cref="Models.Shoko.Renamer"/> by the given <paramref name="renamerID"/>.
+        /// </summary>
+        /// <param name="renamerID">Renamer ID</param>
+        /// <returns></returns>
+        [Authorize("admin")]
+        [HttpPost("{renamerID}/Enable")]
+        public ActionResult EnableRenamerByRenamerID(int renamerID)
         {
-            ServerSettings.Instance.Plugins.EnabledRenamers[renamerID] = false;
+            var renamer = RenameFileHelper.Renamers.Values.FirstOrDefault(r => r.ID == renamerID);
+            if (renamer == null)
+                return NotFound("Renamer not found.");
+
+            ServerSettings.Instance.Plugins.EnabledRenamers[renamer.Name] = true;
             ServerSettings.Instance.SaveSettings();
+
             return Ok();
         }
 
-        [HttpPatch("{renamerID}")]
-        public ActionResult SetPriority(string renamerID, [FromBody] int priority)
+        /// <summary>
+        /// Set the priority of the <see cref="Models.Shoko.Renamer"/> by the given <paramref name="renamerID"/>, optionally enabling it if it's disabled.
+        /// </summary>
+        /// <param name="renamerID">Renamer ID</param>
+        /// <param name="priority">Renamer priority. The lower number the higher the priority.</param>
+        /// <returns></returns>
+        [Authorize("admin")]
+        [HttpPost("{renamerID}/SetPriority")]
+        public ActionResult EditRenamerByRenamerID(int renamerID, [FromQuery] [Range(0, int.MaxValue)] int priority)
```

Comment:
I am looking soon to change fundamentally how we pipeline renamers/scripts, as mentioned in discord